rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,211 @@
+1. Our passenger information (RTR) and dispatch (Glides) applications both implement their own logic to associate vehicles with trips, duplicating effort across teams while also creating additional confusion.
+1. Glides does not publish trainsheet changes using trip IDs that directly correspond to the trip IDs used by GTFS, further complicating the logic in RTR that has to assign trips to vehicles.
+1. Because RTR has to guess as to the correct trip to assign to a train, trip IDs published in our predictions are often not very sensible, or we give up on guessing a scheduled trip completely and assign a train to an added trip. As a result, passenger-facing systems such as mbta.com or screens that may wish to display both scheduled and predicted departure information can't always be sure if two trips with different IDs in the schedule versus the realtime data are actually "the same." Showing scheduled and realtime data for the same trip side-by-side is also a common pattern in third-party applications, so when a trip fails to match to the schedule the result is a combination of a scheduled trip without realtime tracking and a realtime arrival prediction without a corresponding scheduled time, creating a confusing experience for users.
+1. Analysis of historical data for performance purposes is also complicated by the difficulty in linking a trip from the schedule with a trip from the (historical) realtime data.
+
+In addition to the technical challenges, this also creates unclear ownership of different parts of the data pipeline, with different teams implementing their own logic piecemeal to serve their own needs.
+
+## Guiding principles and assumptions
+
+Finally, I want to take a second to lay out some goals that have guided the decisions in this RFC:
+
+1. Having a single source of truth for data.
+1. Creating clear understanding between teams about ownership of different parts of the data pipeline.
+1. Balancing the needs of internal operations users and public-facing users.
+1. Staying as close as possible to the most basic common denominator in data across TID systems, in this case GTFS trips. This means that improvements benefit as many teams and applications as possible, and we adhere to the principle of "small pieces, loosely joined" by a common standard.
+
+The TID data architecture does not exist in a vacuum. For our purposes in this RFC, the main outside factor that we need to be aware of is the HASTUS system used for scheduling and the Plans & Schedules team that uses it to produce the new light rail schedule every rating. Challenges in this area, in particular around schedules during disruptions, will be factored in.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The below diagram gives an overall view of the proposal. Solid lines represent static data, whereas dotted lines represent realtime data:
```

Comment:
question: doesn't HASTUS also feed the OCS trainsheets that feed schedule info to Glides?

Reply:
That arrow does exist for heavy rail but I didn't include it here because OCS doesn't handle light rail trainsheets.